### PR TITLE
Support for px bootstrap install script

### PR DESCRIPTION
**Pull request by zoxpx**
Adding handler for get.portworx.com, which returns the bootstrap install script
for Portworx OCI and Docker.

### CODE REVIEW

File: px-spec-websvc/templates/px-install.sh.tmpl [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+Cyello="$(tput bold 2> /dev/null; tput setaf 3 2> /dev/null)"
+Cnorm="$(tput sgr0 2> /dev/null)"
+
+usage() {
+    cat << _EOF
+Usage: bootstrap-salt.sh [--oci|--docker|--plugin] [--upgrade] <px-options>
+
+Options:
+    --oci      Installs Portworx as OCI Container
+    --docker   Installs Portworx as Docker container (plugin v1)
```
Comment (harsh-px) [inline px-spec-websvc/templates/px-install.sh.tmpl]: I would say let's drop `docker` and `plugin` as options.

File: px-spec-websvc/templates/px-install.sh.tmpl [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    --plugin   Installs Portworx as Docker V2 plugin
+
+    --upgrade  Upgrades existing Portworx OCI Container (OCI only option)
+
+    <px-options>
+               Please see https://docs.portworx.com/runc/#options
+
+Example:
+    bootstrap-salt.sh -c MY_CLUSTER_ID -k etcd://myetc.company.com:2379 -s /dev/xvdb -s /dev/xvdc
```
Comment (harsh-px) [inline px-spec-websvc/templates/px-install.sh.tmpl]: when people curl this, they won't see `bootstrap-salt.sh`. So the script name in usage might be confusing.

File: px-spec-websvc/templates/px-install.sh.tmpl [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+echo "${Cgreen}Installing Portworx as $typeName ...${Cnorm}"
+
+if [ $opt_type -eq 1 ]; then    # install Docker Container
+    opt_img=${opt_img:-portworx/px-enterprise:{{.PxVer}}}
+    sudo docker run --restart=always --name portworx -d --net=host --privileged=true \
+        -v /run/docker/plugins:/run/docker/plugins -v /var/lib/osd:/var/lib/osd:shared \
+        -v /dev:/dev -v /etc/pwx:/etc/pwx -v /opt/pwx/bin:/export_bin \
+        -v /var/run/docker.sock:/var/run/docker.sock -v /var/cores:/var/cores \
+        -v /usr/src:/usr/src $xtravols --ipc=host \
```
Comment (harsh-px) [inline px-spec-websvc/templates/px-install.sh.tmpl]: assuming /usr/src. Doesn't work for coreos. This is low priority as I don't think we even need to support install as docker container.

File: px-spec-websvc/templates/px-install.sh.tmpl [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+Cy="$(tput bold 2> /dev/null; tput setaf 3 2> /dev/null)" # WARN (yellow)
+CR="$(tput sgr0 2> /dev/null)"				  # RESET
+
+usage() {
+    cat << _EOF
+Usage: bootstrap-px.sh [--oci|--docker] [--upgrade] <px-options>
+
+Options:
+    --oci      Installs Portworx as OCI Container (defalt)
+    --docker   Installs Portworx as Docker container
```
Comment (harsh-px) [inline px-spec-websvc/templates/px-install.sh.tmpl]: let's remove docker from all places

File: px-spec-websvc/px-spec-websvc.go [hunk 5]
```diff
@@ -332,9 +364,13 @@ func main() {
 			sendError(http.StatusBadRequest, err, w)
 			return
 		}
-		fmt.Fprintf(w, content)
+		io.WriteString(w, content)
 	})
 
+	// Handler for "get.portworx.com"
+	http.HandleFunc("get.portworx.com/", renderBootstrapScript)
```
Comment (harsh-px) [inline px-spec-websvc/px-spec-websvc.go]: This means the knowledge of where this is running is embedded in the code. We don't want this. Before this, we are able to just deploy the container wherever we want and just change the DNS layer names. The DNS names should just be mentioned in the balancer which in our case is cloudfare.

Instead just create a `/standalone`  path and provide this functionality there. So the usage URL will be install.portworx.com/standalone.

**Comment by zoxpx** [comment]
Hello @harsh-px -- unfortunately I don't agree with your last feedback:
1. I was OK to remove plugin-V2 support, as we stopped building it.  However, I think we are still supporting/building the "standalone docker" install, which is (still) very confusing and obscure for customers to install.  If @gourao and @venkatpx (also possibly Erik and others) agree to remove Docker-support, I'll remove it.
2. That is correct, this installer is supposed to run primarily via the "get.portworx.com" (cloudflare DNS alias) -- exactly the same as get.docker.com, get.saltstack.com and others  (Jeff and Michael already arranged this).  I am OK to set up the bootstrap to run _in addition_ as install.portworx.com/get-px, but I feel strongly about keeping the get.portworx.com.

Since we seem to have ended up in a disagreement around usability / non-technical issues, I think we'll need to include other people to act as tie-breakers into this review.

File: px-spec-websvc/px-spec-websvc.go (hunk 5 above)
Comment (harsh-px) [reply px-spec-websvc/px-spec-websvc.go]: @zoxpx you can keep the user-facing URL as get.portworx.com for standalone installs (so no additional sub-paths in install.portworx.com). But let's not keep the knowledge of that URL in code here. The `http.HandleFunc(` must not be concerned with how people get routed here.

Comment (zoxpx) [reply px-spec-websvc/px-spec-websvc.go]: @harsh-px -- this is how the "virtual host" feature works in golang: the template/code will activate if/when the original request has been made through the "get.portworx.com" host / DNS alias -- therefore we have to put "get.portworx.com" in `http.HandleFunc(`.

Comment (harsh-px) [reply px-spec-websvc/px-spec-websvc.go]: @zoxpx and me synced up on a zoom call. The decision is to go ahead with the current impl.

The other alternative was to have a new web service for this handler and run it on a different aws instance and have the DNS get.portworx.com route to that. Basically, a  parallel pipeline to install.portworx.com route. We are not going ahead with this alternative as with our current team size, we don't want to add additional deployment and maintenance work with a second route.

**Comment by venkatpx** [comment]
@zoxpx I think it is safe to remove docker container option from the standalone install page and only offer OCI. IMHO, we need to start pushing customers towards OCI

**Comment by zoxpx** [comment]
Adding @jrivera-px to review (sorry -- forgot to include earlier).

**Pull request merged**